pkg/apiserver/handlers/webhook/mutation_labels_test.go

```diff
@@ -0,0 +1,257 @@
+		isAllowed bool
+		msg       string
+		expResp   []jsonPatch
+	}{
+		{
+			name:      "mutate-labels-empty-create",
+			op:        admv1.Create,
+			resName:   "my-ns",
+			labels:    emptyLabels,
+			isAllowed: true,
+			msg:       "",
+			expResp: []jsonPatch{
+				{
+					Op:    jsonPatchReplaceOp,
+					Path:  labelsPath,
+					Value: nameLabel,
+				},
+			},
+		},
+		{
+			name:      "mutate-labels-empty-update",
+			op:        admv1.Update,
+			resName:   "my-ns",
+			labels:    emptyLabels,
+			isAllowed: true,
+			msg:       "",
+			expResp: []jsonPatch{
+				{
+					Op:    jsonPatchReplaceOp,
+					Path:  labelsPath,
+					Value: nameLabel,
+				},
+			},
+		},
+		{
+			name:      "mutate-labels-empty-delete",
+			op:        admv1.Delete,
+			resName:   "my-ns",
+			labels:    emptyLabels,
+			isAllowed: true,
+			msg:       "",
+			expResp:   []jsonPatch{},
+		},
+		{
+			name:      "mutate-labels-random-create",
+			op:        admv1.Create,
+			resName:   "my-ns",
+			labels:    randomLabel,
+			isAllowed: true,
+			msg:       "",
+			expResp: []jsonPatch{
+				{
+					Op:    jsonPatchReplaceOp,
+					Path:  labelsPath,
+					Value: randomLabelPlusNameLabel,
+				},
+			},
+		},
+		{
+			name:      "mutate-labels-random-update",
+			op:        admv1.Update,
+			resName:   "my-ns",
+			labels:    randomLabel,
+			isAllowed: true,
+			msg:       "",
+			expResp: []jsonPatch{
+				{
+					Op:    jsonPatchReplaceOp,
+					Path:  labelsPath,
+					Value: randomLabelPlusNameLabel,
+				},
+			},
+		},
+		{
+			name:      "mutate-labels-random-delete",
+			op:        admv1.Delete,
+			resName:   "my-ns",
+			labels:    randomLabel,
+			isAllowed: true,
+			msg:       "",
+			expResp:   []jsonPatch{},
+		},
+		{
+			name:      "mutate-labels-existing-name-label-create",
+			op:        admv1.Create,
+			resName:   "my-ns",
+			labels:    nameLabel,
+			isAllowed: true,
+			msg:       "",
+			expResp: []jsonPatch{
+				{
+					Op:    jsonPatchReplaceOp,
+					Path:  labelsPath,
+					Value: nameLabel,
+				},
+			},
+		},
+		{
+			name:      "mutate-labels-existing-name-label-update",
+			op:        admv1.Update,
+			resName:   "my-ns",
+			labels:    nameLabel,
+			isAllowed: true,
+			msg:       "",
+			expResp: []jsonPatch{
+				{
+					Op:    jsonPatchReplaceOp,
+					Path:  labelsPath,
+					Value: nameLabel,
+				},
+			},
+		},
+		{
+			name:      "mutate-labels-existing-name-label-delete",
+			op:        admv1.Delete,
+			resName:   "my-ns",
+			labels:    nameLabel,
+			isAllowed: true,
+			msg:       "",
+			expResp:   []jsonPatch{},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var expValue []byte
+			actualMsg, actualAllowed, actualResp := mutateLabels(tt.op, tt.labels, tt.resName)
+			// We dont expect Delete operations to return any patch response.
+			if tt.op != admv1.Delete {
+				expValue, _ = json.Marshal(tt.expResp)
+			}
+			if bytes.Compare(expValue, actualResp) != 0 {
```

Comment:
there is a `bytes.Equal` function, but I thought that `assert.Equal` worked in that case?

Reply:
done

Reply:
done